iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerGrammar.java

```diff
@@ -342,7 +343,8 @@ public ShellForm SHELL_FORM_GENERIC() {
     return b.<ShellForm>nonterminal(DockerLexicalGrammar.SHELL_FORM_GENERIC).is(
       f.shellForm(
         b.oneOrMore(
-          f.ignoreFirst(b.token(DockerLexicalGrammar.WHITESPACE), ARGUMENT_GENERIC()))));
+          f.ignoreFirst(b.token(DockerLexicalGrammar.WHITESPACE), ARGUMENT_GENERIC())),
+        b.optional(f.hereDocument(b.token(DockerLexicalGrammar.HEREDOC_EXPRESSION)))));
   }
 
   public HereDocument HEREDOC_FORM() {
```

Comment:
Maybe rename the occurrences of `HereDocForm` to `HereDoc`, as we are not considering it anymore as a form like `ShellForm` or `ExecForm` but as content that is part of `ShellForm`.